csharp/src/Ice/OutputStream.cs

```diff
@@ -1985,29 +1986,28 @@ private void RewriteByteSpan(Span<byte> data, Position pos)
             }
         }
 
-        /// <summary>Returns the current position and writes a 4-bytes placeholder for a fixed-length size value. The
+        /// <summary>Returns the current position and writes placeholder for a fixed-length size value. The
         /// position must be used to rewrite the size later.</summary>
+        /// <param name="sizeLenght">The number of bytes reserved to write the fixed-length size.</param>
         /// <returns>The position before writing the size.</returns>
-        private Position StartFixedLengthSize()
+        internal Position StartFixedLengthSize(int sizeLenght = DefaultSizeLength)
```

Comment:
I think all these methods should remain private. We could however add one or more helper internal methods to OutputStream to help with the binary context marshaling.